### PR TITLE
Add documentation for module mounts

### PR DESCRIPTION
**Pull request by jedesah**

### CODE REVIEW

**Comment by alissapajer** [comment]
Restarted failed travis job (even though this is just a doc change). Error was the one fixed in #2111.

**Comment by jedesah** [comment]
Just realized it's a little unreasonable to expect @beckyconning to merge this herself considering we use a script for merging. @sellout Can you merge this if @beckyconning signs off on it actually being clear and useful?

**Comment by jedesah** [comment]
@beckyconning I know you are probably sure busy these days but if you could just give a :+1: on this PR or mention something that could be improved

File: README.md [hunk 1]
```diff
@@ -294,6 +294,22 @@ For example, given the above MongoDB mount, an additional view could be defined
...
+The value of a module mount is simply the SQL string which will be parsed into a collection of SQL Statements.
+
+To create a new module one would send a json blob similar to this one to the mount endpoint:
+
+```json
+{ "module": "CREATE FUNCTION ARRAY_LENGTH(:foo) BEGIN COUNT(:foo[_]) END; CREATE FUNCTION USER_DATA(:user_id) BEGIN SELECT * FROM `/root/path/data/` WHERE user_id = :user_id END" }
+```
+
+Similar to views, modules can be mounted at any directory path. If a module's path is nested inside the path of a database mount, it will appear alongside the other directory and files in the database. A module will "shadow" any actual directory that would otherwise be mapped to the same path. Any attempt to write data to a module will result in an error.
+
```
Comment (natefaubion) [inline README.md]: Are function names like identifiers (case sensitive), or are they like builtins (case insensitive)?

Comment (jedesah) [inline README.md]: Like builtins (case insensitive). Will add a line with that information.

Comment (natefaubion) [inline README.md]: In SQL it is insensitive, but in my testing, the URL is case sensitive. It would be worth clarifying that.

**Comment by sellout** [comment]
@beckyconning Just a pink so you can indicate whether this PR meets your needs.

File: README.md (hunk 1 above)
Comment (jedesah) [inline README.md]: Addressed this issue [here](https://github.com/quasar-analytics/quasar-docs/pull/2) with a reference in this README.

**Comment by jedesah** [comment]
@sellout Actually I switched over responsibility to @natefaubion for critiquing this PR which he did and I've made for updates to reflect them. So hopefully @natefaubion you can have another look at it.

I'll update the documentation on `/invoke` in the PR that add support for `offset` and `limit`.

**Comment by sellout** [comment]
@natefaubion Ping.

**Comment by alissapajer** [comment]
The build is now green for this. @natefaubion is this ready for merge?

**Comment by jedesah** [comment]
@alissapajer I think @natefaubion already approved. I think this is just waiting for @sellout to merge

**Comment by sellout** [comment]
Ah, as usual – need to add comments when you do stuff on GitHub, because very few things actually trigger a notification.

**Pull request merged**